src/Connection/ImapConnection.php

```diff
@@ -701,7 +701,8 @@ protected function newTokenizer(StreamInterface $stream): ImapTokenizer
      */
     protected function assertTaggedResponse(string $tag, ?callable $exception = null): TaggedResponse
     {
-        return $this->assertNextResponse(
+        /** @var TaggedResponse $response */
```

Comment:
I'm a bit unhappy with this one. Everything says It is "TaggedResponse", but the method signature of assertNextResponse is too complex for PHPStan to understand it directly. This way we do a type coercion and return it afterwards. I don't see a cleaner solution for this, without significantly changing how assert*Response works.

Reply:
This is totally fine with me! 👍